doc/source/contributing.md

```diff
@@ -0,0 +1,87 @@
+# Contributing to `xyzservices`
+
+Contributions to `xyzservices` are very welcome. They are likely to be accepted more
+quickly if they follow these guidelines.
+
+There are two main groups of contributions - adding new provider sources and
+contributions to the codebase and documentaion.
+
+## Providers
+
+If you want to add a new provider, simply add its details to
+`provider_sources/xyzservices-providers.json` and then finalise the JSON using:
+
+```
+make compress
+```
+
+You can add a single `TileProvider` or a `Bunch` of `TileProviders` using the following
+schema:
+
+```json
+{
+   "single_provider_name": {
+      "url": "https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png",
+      "max_zoom": 19,
+      "attribution": "(C) OpenStreetMap contributors",
+      "name": "OpenStreetMap.Mapnik"
+   },
+   "provider_bunch_name": {
+      "first_provider_name": {
+            "url": "https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png",
+            "max_zoom": 19,
+            "attribution": "(C) OpenStreetMap contributors",
+            "name": "OpenStreetMap.Mapnik"
+      },
+      "second_provider_name": {
+            "url": "https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png?access-token={accessToken}",
+            "max_zoom": 19,
+            "attribution": "(C) OpenStreetMap contributors",
+            "name": "OpenStreetMap.Mapnik",
+            "accessToken": "<insert your access token here>"
+      }
+   }
+}
+```
```

Comment:
I'd use a single one. My suggestion would be a single source per PR? Unless they are related (e.g. Toner, TonerLite, TonerNoLabels). What do you think?